test-optimizations.sh

```
@@ -0,0 +1,171 @@
+# Test 2: Check .dockerignore effectiveness
+echo -e "${BLUE}✅ Test 2: Build context size check${NC}"
+context_line=$(docker build --no-cache -t desci-server:context-test . 2>&1 | grep "Sending build context" | head -1 || echo "Sending build context unknown")
+echo "Build context: $context_line"
+
+# Extract size if possible
+if [[ $context_line =~ ([0-9.]+[KMGT]?B) ]]; then
+    size="${BASH_REMATCH[1]}"
+    echo -e "${GREEN}✓${NC} Context size: $size"
+    
+    # Check if size seems reasonable (less than 200MB)
+    if [[ $context_line =~ ([0-9.]+)MB && ${BASH_REMATCH[1]} < 200 ]] || [[ $context_line =~ ([0-9.]+)[KB] ]]; then
```

Shellcheck (0.10.0)

[error] < is for string comparisons. Use -lt instead.

(SC2071)